plugins/modules/omapi_host.py

```diff
@@ -231,8 +234,8 @@ def setup_host(self):
             response_obj = self.unpack_facts(host_response.obj)
             fields_to_update = {}
 
-            if to_bytes('ip-address', errors='surrogate_or_strict') not in response_obj or \
-                    unpack_ip(response_obj[to_bytes('ip-address', errors='surrogate_or_strict')]) != self.module.params['ip']:
+            if 'ip-address' not in response_obj or \
+                    unpack_ip(response_obj['ip-address']) != self.module.params['ip']:
```

Comment:
Hmm, I wonder whether this is correct. `response_obj['ip-address']` has already been unpacked in `self.unpack_facts()`, I'm not sure what will happen if it is unpacked again?

Reply:
Good eye! I’ll drop that as well. (This may even have been the reason why updating didn’t always work…)